plone/app/caching/purge.py

```diff
@@ -233,22 +233,28 @@ def purgeOnModified(object, event):
 
 
 @adapter(IContentish, IObjectMovedEvent)
-def purgeOnMovedOrRemoved(object, event):
-    request = getRequest()
-    confirmed_delete = (
-        request is not None
-        and getattr(request, "URL", None)
-        and "delete_confirmation" in request.URL
-        and request.REQUEST_METHOD == "POST"
-        and "form.submitted" in request.form
-    )
-    if not confirmed_delete and IObjectRemovedEvent.providedBy(event):
-        # ignore extra delete events
+def purgeOnMoved(object, event):
+    """Purge after object was moved."""
+
+    # Purge on removal is handled elsewhere
+    if IObjectRemovedEvent.providedBy(event):
         return
     # Don't purge when added
     if IObjectAddedEvent.providedBy(event):
         return
-    if isPurged(object) and "portal_factory" not in request.URL:
+    # check to not include move-events during content creation
+    if isPurged(object):
+        notify(Purge(object))
+
+    parent = object.getParentNode()
+    if parent:
```

Comment:
Why isn't the parent's purgeability checked? Like
```
if parent and isPurged(parent):
```

Reply:
I forgot! It is added now.

Reply:
@pgrunewald, actually, you didn’t forget it. It's just that in the previous code, that check wasn’t there at all... I think it was a mistake; I don't see a reason to trigger a purge on content that isn’t defined as purgeable...

Reply:
You're are corrected. I also think it's better to check the purgeability before actually purging.